### ISSUE TITLE
add_to_app books sample cannot resolve mockito dependency

### ISSUE THREAD

**Issue by bryanoltman** [flutter/samples#877]
When running `flutter pub get` from the `add_to_app/books` directory, it fails with the following error:

```
Because no versions of mockito match >4.1.1+1 <4.1.2 and mockito >=4.0.0 <=4.1.1 depends on test_api ^0.2.1, mockito >=4.0.0 <4.1.1+1 or >4.1.1+1 <4.1.2-∞ requires test_api ^0.2.1.
And because mockito >=4.1.2 <=5.0.0-nullsafety.7 depends on test_api ^0.2.19-nullsafety, mockito >=4.0.0 <4.1.1+1 or >4.1.1+1 <5.0.0-∞ requires test_api ^0.2.1.
And because mockito 4.1.1+1 depends on test_api >=0.2.1 <0.4.0 and every version of flutter_test from sdk depends on test_api 0.4.2, flutter_test from sdk is incompatible with mockito >=4.0.0 <=5.0.0-nullsafety.7.
So, because flutter_module_books depends on both mockito ^4.1.1 and flutter_test any from sdk, version solving failed.
Running "flutter pub get" in flutter_module_books...
pub get failed (1; So, because flutter_module_books depends on both mockito ^4.1.1 and flutter_test any from sdk, version solving failed.)
```

**Comment by jmagman** [comment]
It seems to be working in CI https://github.com/flutter/samples/runs/3366765137?check_suite_focus=true#step:5:123
Can we get the output of `flutter doctor -v`?

**Comment by bryanoltman** [comment]
Sure thing!

```
[✓] Flutter (Channel master, 2.4.0-5.0.pre.59, on macOS 11.5.1 20G80 darwin-x64, locale en-US)
    • Flutter version 2.4.0-5.0.pre.59 at /Users/bryanoltman/Documents/flutter
    • Upstream repository git@github.com:bryanoltman/flutter.git
    • Framework revision f71ceb77fe (5 weeks ago), 2021-07-15 10:18:49 -0700
    • Engine revision 093c4ed1a3
    • Dart version 2.14.0 (build 2.14.0-312.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc1)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2021.1.3)
    • IntelliJ at /Applications/IntelliJ CE (stable).app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 8F94B6B5-06A6-4E8A-BF78-E80698729705 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 92.0.4515.159

! Doctor found issues in 2 categories.
```

There are a couple issues, but not anything I would expect to generate the error I'm seeing.

**Comment by jmagman** [comment]
See https://github.com/flutter/flutter/issues/82280 for more context.

**Comment by RedBrogdon** [comment]
This has been resolved in #895, by removing the Mockito dependency entirely.
